### PR TITLE
history table

### PR DESCRIPTION
**Pull request by Djoudi21**
table implemented
data fetching  implemented
restoring by unit  implemented
restoring by bulk implemented
tables filtering implemented
grid row and search bar components extract
coming soon style added to delete icon

### CODE REVIEW

File: src/api/ropa/RopaArchiveRepository.ts [hunk 1]
```diff
@@ -19,4 +19,12 @@ export class RopaArchiveRepository {
     // TODO: When the response will contain a message key, update the object below to use it instead of res.statusText
     return { code: res.status, message: res.statusText }
   }
+
+  async restoreById(id: number) {
+    function generateUrl(id) {
```
Comment (Jabuf) [inline src/api/ropa/RopaArchiveRepository.ts:24]: I'm not sure why we are using a function here ? Below the same code rewritten without a function, it seems more readable and easier to me , but maybe I'm missing something ?

```
    const url = ApiPaths.putRestoreRopaById.replace(':ropaId', id.toString())
    await this.ropaHttpClient.put(url)
```

File: src/api/ropa/ropaPagesUropaDashboardRepository.ts [hunk 2]
```diff
@@ -25,4 +32,18 @@ export class RopaPagesUropaDashboardRepository {
       res
     )
   }
+
+  async getHistoryByCatalogId(
+    catalogId: number
+  ): Promise<GetPagesUropaDashboardHistoryByCatalogIdRes> {
+    function generateUrl(catalogId) {
+      return ApiPaths.getPagesUropaDashboardHistory.replace(
```
Comment (Jabuf) [inline src/api/ropa/ropaPagesUropaDashboardRepository.ts:40]: Same remark as above.

File: src/api/ropa/ropaRopaRepository.ts [hunk 3]
```diff
@@ -60,4 +62,16 @@ export class RopaRopaRepository {
     )
     return extractEntityFromRopaRes<PutRopaRopaByIdRes>(res)
   }
+
+  /**
+   * @deprecated
+   * @param id
+   */
+  async delById(id: number): Promise<DelRopaRopaByRes> {
```
Comment (Jabuf) [inline src/api/ropa/ropaRopaRepository.ts:70]: Nothing big, but personally not a big fan of using a shortened name here. I think that deleteById is more explicit and easy to read.

File: src/data/adapters/PagesUropaDashboardAdapter.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { ropaApi } from '@/api/ropa/ropaApi'
+
+export abstract class PagesUropaDashboardAdapter {
+  static async getById(catalogId: number) {
```
Comment (Jabuf) [inline src/data/adapters/PagesUropaDashboardAdapter.ts:4]: Again, nothing big, but I don't a `getById` naming is appropriate here. Indeed a dashboard isn't a resource that we can fetch using its id, the id here is the `catalogId`. I think `getByCatalogId` is more explicit and you immediately know what to use.

Comment (Jabuf) [reply src/data/adapters/PagesUropaDashboardAdapter.ts:4]: I've just seen the function above. If this adapter can return two different type of contents, then it make sense to differentiate them in the naming of function (exactly how it was done below). So `getHistoryByCatalogId` and `getDashboardByCatalogId`.

**Review by Jabuf — COMMENTED** [review]
Some remarks about the naming and syntax, but nothing preventing the merge.

File: src/api/ropa/ropaRopaRepository.ts (hunk 3 above)
Comment (mrz1880) [reply src/api/ropa/ropaRopaRepository.ts:70]: I agree, we will rename it.

File: src/api/ropa/ropaPagesUropaDashboardRepository.ts (hunk 2 above)
Comment (mrz1880) [reply src/api/ropa/ropaPagesUropaDashboardRepository.ts:40]: Yes totally ! Initially, I was thinking about centralize the logic but the logic is too simple to be abstracted.

File: src/api/ropa/RopaArchiveRepository.ts (hunk 1 above)
Comment (mrz1880) [reply src/api/ropa/RopaArchiveRepository.ts:24]: Yes totally ! Initially, I was thinking about centralize the logic but the logic is too simple to be abstracted.

**Pull request merged**